packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -192,11 +199,29 @@ export function createInsightsMiddleware<
           instantSearchInstance.emit('error', new Error(errorMessage));
         }
       },
+      // eslint-disable-next-line complexity
```

Comment:
what is the complexity about? can it be avoided?

Reply:
I'll try to extract some functionality into separate functions, should fix this.